src/Api/Parser/JsonParser.php

```diff
@@ -17,11 +17,18 @@ public function parse(Shape $shape, $value)
 
         switch ($shape['type']) {
             case 'structure':
+                if (isset($shape['document']) && $shape['document']) {
+                    return $value;
+                }
                 $target = [];
                 foreach ($shape->getMembers() as $name => $member) {
                     $locationName = $member['locationName'] ?: $name;
                     if (isset($value[$locationName])) {
-                        $target[$name] = $this->parse($member, $value[$locationName]);
+                        if ($shape->getMember($locationName)['document']) {
```

Comment:
You shouldn't be looking up a shape by location name, but the actual member name. Also, isn't this handled properly by a recursive call to `parse` anyways? I'm not sure this change is necessary.

Reply:
I removed this without any apparent change in functionality- good catch